### PR TITLE
It's the work for the final project from Yamabukiss

### PR DESCRIPTION
**Pull request by Yamabukiss**
The changes of the repositories will be list underline:

There will contains 3 part:
1:
the conclusion of my summary of the class   
//
in dir : missing-semester-1/2021-finalproject/Yamabukiss   
the file README.md 
//

2:
the test scripts and the screenshot of the test of scripts's result
//
in dir : missing-semester-1/2021-finalproject/Yamabukiss   
the file testProgram.sh
//

3:
the application of the git 
//
in dir : missing-semester-1/2021-finalproject/Yamabukiss /git application
//

### CODE REVIEW

**Review by Yamabukiss — COMMENTED** [review]
It's the review from Yamabukiss

**Comment by Yamabukiss** [comment]
there are some changes needed your reviewing
please check it
@MuZhou233

File: 2021-finalproject/Yamabukiss/git application/回退修改.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+1、若你已经修改了部分文件、并且将其中的一部分加入了暂存区，应该如何回退这些修改，恢复到修改前最后一次提交的状态？给出至少两种不同的方式
```
Comment (MuZhou233) [inline 2021-finalproject/Yamabukiss/git application/回退修改.md:1]: Don't use chinese in file name

File: 2021-finalproject/Yamabukiss/testProgram.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+else 
+	echo "the ip address is available"
+fi
+
+# judge if the local ip is in the same subnet with target ip
+echo "local ip configuration:"
+ifconfig | awk '/netmask/{print $0}'  | awk 'NR==1'
+echo "target ip is:"
+echo $ip
+echo "they are't in the same subnet"
```
Comment (MuZhou233) [inline 2021-finalproject/Yamabukiss/testProgram.sh:36]: It seems you lost judge sentences here

**Review by MuZhou233 — APPROVED** [review]

File: 2021-finalproject/Yamabukiss/testProgram.sh (hunk 2 above)
Comment (Yamabukiss) [reply 2021-finalproject/Yamabukiss/testProgram.sh:36]: actually i think it's clear to see by the compare  so i have't set the judge sentences should i add it into the scripts?

Comment (Yamabukiss) [reply 2021-finalproject/Yamabukiss/testProgram.sh:36]: @MuZhou233

Comment (MuZhou233) [reply 2021-finalproject/Yamabukiss/testProgram.sh:36]: If so, tell user attention to the comparation is okay.
Or you can try to add judge and avoid print the detials.

Comment (Yamabukiss) [reply 2021-finalproject/Yamabukiss/testProgram.sh:36]: i see

Comment (Yamabukiss) [reply 2021-finalproject/Yamabukiss/testProgram.sh:36]: > If so, tell user attention to the comparation is okay. Or you can try to add judge and avoid print the detials.

i see

**Comment by Yamabukiss** [comment]
i have add some changes in the program 
please check it and tell me if there is still something need to changed 
@MuZhou233

**Comment by MuZhou233** [comment]
> i have add some changes in the program please check it and tell me if there is still something need to changed

Add a screenshot for a full pass test case

**Comment by Yamabukiss** [comment]
> > i have add some changes in the program please check it and tell me if there is still something need to changed
> 
> ‎添加所有通过测试的屏幕截图‎case



> > i have add some changes in the program please check it and tell me if there is still something need to changed
> 
> Add a screenshot for an all pass test case

added yet 
under dir:Yamabukiss

**Comment by MuZhou233** [comment]
> added yet under dir:Yamabukiss

We need two screenshot here, via https://github.com/gdut-dynamic-x/missing-semester/blob/ca6d27b4ef3318474fa3944239d52df6296df10d/2021-finalproject/README.md?plain=1#L37

**Comment by Yamabukiss** [comment]
> > added yet under dir:Yamabukiss
> 
> ‎我们需要两个截图‎‎路‎https://github.com/gdut-dynamic-x/missing-semester/blob/ca6d27b4ef3318474fa3944239d52df6296df10d/2021-finalproject/README.md?plain=1#L37



> > added yet under dir:Yamabukiss
> 
> We need two screenshot here, via https://github.com/gdut-dynamic-x/missing-semester/blob/ca6d27b4ef3318474fa3944239d52df6296df10d/2021-finalproject/README.md?plain=1#L37

i don't actually know the means of the sentence ,is that means screenshots of pass and not pass?

**Comment by Yamabukiss** [comment]
> > added yet under dir:Yamabukiss
> 
> We need two screenshot here, via https://github.com/gdut-dynamic-x/missing-semester/blob/ca6d27b4ef3318474fa3944239d52df6296df10d/2021-finalproject/README.md?plain=1#L37

i have add the pass screenshot of scripts

**Review by MuZhou233 — DISMISSED** [review]

**Comment by Yamabukiss** [comment]
please merge my request @MuZhou233

**Comment by MuZhou233** [comment]
Although the path-check action shows passed, your changes are not conform to our newest check item. If you push a new commit, you will see that check action failed.

**Comment by Yamabukiss** [comment]
@MuZhou233 rename the directory yet

File: _2021-finalproject/README.md [hunk 3]
```diff
@@ -1,63 +1,64 @@
-# Final Project
```
Comment (MuZhou233) [inline _2021-finalproject/README.md:1]: Your commit sholdn't touch this file

**Review by MuZhou233 — CHANGES_REQUESTED** [review]

**Comment by Yamabukiss** [comment]
there is confict exsit so i replace the README.md file by the new one @MuZhou233

**Comment by MuZhou233** [comment]
Why your commits delete submitted files

**Comment by Yamabukiss** [comment]
> Why your commits delete submitted files
well  i have pulled the repository in the wrong way so i reset the repository

**Comment by Yamabukiss** [comment]
is there still something wrong? 
i don't get it. i think I've done it : )

**Comment by MuZhou233** [comment]
As I said, your commits sholdn't delete other files. check this and think what you are doing https://github.com/gdut-dynamic-x/missing-semester/commits/master

**Comment by Yamabukiss** [comment]
now? @MuZhou233

**Comment by MuZhou233** [comment]
Double check the file changed list

**Comment by Yamabukiss** [comment]
and? @MuZhou233

**Comment by MuZhou233** [comment]
Do the same thing to `_2021-finalproject/LYD-cubing/text.sh` and `_2021-finalproject/TNTksals/scripts/check-environment.sh`. You know, that's not your file.

**Comment by Yamabukiss** [comment]
@MuZhou233  ihave change the path  why i can't pass?

**Comment by Yamabukiss** [comment]
@MuZhou233 it's done

**Comment by Yamabukiss** [comment]
> @MuZhou233 it's done

QAQ please review it @MuZhou233

File: _2021-finalproject/README.md (hunk 3 above)
Comment (MuZhou233) [reply _2021-finalproject/README.md:1]: I really appreciate your efforts. But this appears again.

Comment (MuZhou233) [reply _2021-finalproject/README.md:1]: Maybe this is your last work to pass checks.

**Review by MuZhou233 — CHANGES_REQUESTED** [review]

File: _2021-finalproject/README.md (hunk 3 above)
Comment (Yamabukiss) [reply _2021-finalproject/README.md:1]: > Maybe this is your last work to pass checks.
the information here should't exit?

**Comment by Yamabukiss** [comment]
it's done @MuZhou233

**Comment by Yamabukiss** [comment]
@MuZhou233 please review it 👍

**Review by MuZhou233 — APPROVED** [review]

**Pull request merged**